File: AI.Core/OperatorsImplimentation/MetaOperators/IsChannelExists.cs
using System;
using AI.Core.ChannelsImplimentation;

namespace AI.Core.OperatorsImplimentation.MetaOperators
{
    class IsChannelExists : ChannelOperator
    {
        /// <summary>
        /// Returns "1" if channel between two Entities(with respect to contact number) exists and "0" if not.
        /// </summary>
        public IsChannelExists(Entity thisEntity)
            : base(thisEntity){}

        public override void Action()
        {
            bool result = ThisEntity.ChannelsTable.ContainsKey(new ChannelTableKey(FromOperator, ToOperator));
            ExitContacts.SetValue(Convert.ToUInt64(result));
        }
    }
}

File: NativeCore/OperatorsImplimentation/MetaOperators/IsChannelExists.cpp
#include "IsChannelExists.h"
#include "ChannelsImplimentation/ChannelsTable.h"

//C# TO C++ CONVERTER TODO TASK: The .NET System namespace is not available from native C++:
//using namespace System;
using namespace AI::Core::ChannelsImplimentation;
namespace AI
{
	namespace Core
	{
		namespace OperatorsImplimentation
		{
			namespace MetaOperators
			{

				IsChannelExists::IsChannelExists(Entity *thisEntity) : ChannelOperator(thisEntity)
				{
				}

				void IsChannelExists::Action()
				{
					bool result = getThisEntity()->getChannelsTable()->ContainsKey(ChannelTableKey(getFromOperator(), getToOperator()));
					ExitContacts->SetValue(static_cast<unsigned long long>(result));
				}
			}
		}
	}
}
